### PR TITLE
2 tx flow, first remove votes and undelegate then delegate

### LINKED ISSUE #167: Remove the vote card and add additional info when delegating
As per @goldsteinsveta's idea, the vote card is kindda weird, as Delegit is about delegation rather than votes
we could:
- remove the vote card
- add some more info the the Alert info under the delegation conviction saying
  - what tracks we're about to un-delegate
  - what votes we're about to remove

### LINKED ISSUE #185: Separate the delegate and undelegate in 2 batchs
because of exhaustedressources and fees issues.

### PR DESCRIPTION
**Pull request by Tbaut**
closes #185 
closes #167 

image

---

Submission checklist:

#### Layout

- [x] Change inspected in the mobile web ui
- [x] Change inspected in dark mode

### CODE REVIEW

File: src/components/LocksCard.tsx [hunk 1]
```diff
@@ -273,20 +268,9 @@ export const LocksCard = () => {
               </>
             </ContentReveal>
           </Card>
-          <Card className="h-max border-2 p-2 px-4">
+          {/* <Card className="h-max border-2 p-2 px-4">
```
Comment (wirednkod) [inline src/components/LocksCard.tsx:271]: Please remove commented out code

File: src/components/LocksCard.tsx [hunk 2]
```diff
@@ -315,7 +299,7 @@ export const LocksCard = () => {
                 })}
               </ContentReveal>
             }
-          </Card>
+          </Card> */}
```
Comment (wirednkod) [inline src/components/LocksCard.tsx:302]: Please remove commented out code

File: src/pages/Delegate/MultiTransactionDialog.tsx [hunk 3]
```diff
@@ -147,9 +147,9 @@ export const MultiTransactionDialog = ({
           <DialogDescription>
             <div className="my-4">
               {step === 1 &&
-                'The delegation process is in 2 parts. First, please sign a transaction to remove current votes and delegations.'}
+                'The delegation process is in 2 parts. First, please sign a transaction to remove previous votes or delegations.'}
```
Comment (wirednkod) [inline src/pages/Delegate/MultiTransactionDialog.tsx:150]:
```suggestion
                'Please sign a transaction to clear any previous votes or delegations before proceeding with the delegation process.'}
```

**Review by wirednkod — CHANGES_REQUESTED** [review]

File: src/components/Alert.tsx [hunk 4]
```diff
@@ -1,16 +1,17 @@
 import { Alert, AlertDescription, AlertTitle } from '@/components/ui/alert'
-import { AlertCircle } from 'lucide-react'
+import { AlertCircle, InfoIcon } from 'lucide-react'
 
 interface Props {
-  title: string
-  message: string
+  title?: string
+  message: string | React.ReactNode
```
[the remark this answers is not in the document]
Comment (Tbaut) [reply src/components/Alert.tsx:6]: Nah! this is pretty common and doesn't change anything in the code below. Having 2 variables makes it uglier, and doesn't change anything! For instance that's what we have in the `ContentReveal` and what many common libraries have too!

File: src/components/LocksCard.tsx (hunk 1 above)
Comment (Tbaut) [reply src/components/LocksCard.tsx:271]: oh, forgot about it thx

File: src/pages/Delegate/MultiTransactionDialog.tsx (hunk 3 above)
Comment (Tbaut) [reply src/pages/Delegate/MultiTransactionDialog.tsx:150]: I find this long phrase that we had actually pretty bad, nobody reads such long things, it needs to be more actionable IMHO. What do you think about:
```
There will be 2 transactions:
1. Clear previous votes & delegations
2. Delegate
```

Comment (Tbaut) [reply src/pages/Delegate/MultiTransactionDialog.tsx]: I went for something straight forward. No need to "Please" and what not, they are asking for it, not delegit!
- First, let's clear previous votes or delegations.
- Now, we are ready to delegate.

WDYT?

**Comment by Tbaut** [comment]
ping @wirednkod for another review please :pray:

File: src/pages/Delegate/MultiTransactionDialog.tsx (hunk 3 above)
Comment (wirednkod) [reply src/pages/Delegate/MultiTransactionDialog.tsx]: Sounds better.

**Review by wirednkod — APPROVED** [review]

**Pull request merged**